code/modules/mod/mod_core.dm

```diff
@@ -330,11 +330,12 @@
 		return COMPONENT_NO_AFTERATTACK
 	return NONE
 
-/obj/item/mod/core/plasma/proc/charge_plasma(obj/item/attacking_item, mob/user)
-	if(!istype(attacking_item, /obj/item/stack/ore/plasma))
+/obj/item/mod/core/plasma/proc/charge_plasma(obj/item/stack/plasma, mob/user)
+	if(!istype(plasma))
 		return FALSE
-	if(!attacking_item.use(1))
+	var/uses_needed = min(plasma.amount, round((max_charge_amount() - charge_amount()) / charge_given))
+	if(!plasma.use(uses_needed))
```

Comment:
If you don't have enough plasma to fill it all in one go, is it just impossible to refill it?